packages/loot-core/src/client/queries/queriesSlice.ts

```diff
@@ -0,0 +1,867 @@
+  name: CategoryGroupEntity['name'];
+};
+
+export const createGroup = createAppAsyncThunk(
+  `${sliceName}/createGroup`,
+  async ({ name }: CreateGroupPayload, { dispatch }) => {
+    const id = await send('category-group-create', { name });
+    dispatch(getCategories());
+    return id;
+  },
+);
+
+type UpdateGroupPayload = {
+  group: CategoryGroupEntity;
+};
+
+export const updateGroup = createAppAsyncThunk(
+  `${sliceName}/updateGroup`,
+  async ({ group }: UpdateGroupPayload, { dispatch }) => {
+    // Strip off the categories field if it exist. It's not a real db
+    // field but groups have this extra field in the client most of the time
+    const { categories: _, ...groupNoCategories } = group;
+    await send('category-group-update', groupNoCategories);
+    await dispatch(getCategories());
+  },
+);
+
+type DeleteGroupPayload = {
+  id: CategoryGroupEntity['id'];
+  transferId?: CategoryGroupEntity['id'];
+};
+
+export const deleteGroup = createAppAsyncThunk(
+  `${sliceName}/deleteGroup`,
+  async ({ id, transferId }: DeleteGroupPayload, { dispatch }) => {
+    await send('category-group-delete', { id, transferId });
+    await dispatch(getCategories());
+    // See `deleteCategory` for why we need this
+    await dispatch(getPayees());
+  },
+);
+
+type CreateCategoryPayload = {
+  name: CategoryEntity['name'];
+  groupId: CategoryGroupEntity['id'];
+  isIncome: boolean;
+  isHidden: boolean;
+};
+export const createCategory = createAppAsyncThunk(
+  `${sliceName}/createCategory`,
+  async (
+    { name, groupId, isIncome, isHidden }: CreateCategoryPayload,
+    { dispatch },
+  ) => {
+    const id = await send('category-create', {
+      name,
+      groupId,
+      isIncome,
+      hidden: isHidden,
+    });
+    await dispatch(getCategories());
+    return id;
+  },
+);
+
+type UpdateCategoryPayload = {
+  category: CategoryEntity;
+};
+
+export const updateCategory = createAppAsyncThunk(
+  `${sliceName}/updateCategory`,
+  async ({ category }: UpdateCategoryPayload, { dispatch }) => {
+    await send('category-update', category);
+    await dispatch(getCategories());
+  },
+);
+
+type DeleteCategoryPayload = {
+  id: CategoryEntity['id'];
+  transferId?: CategoryEntity['id'];
+};
+
+export const deleteCategory = createAppAsyncThunk(
+  `${sliceName}/deleteCategory`,
+  async ({ id, transferId }: DeleteCategoryPayload, { dispatch }) => {
+    const { error } = await send('category-delete', { id, transferId });
+
+    if (error) {
+      switch (error) {
+        case 'category-type':
+          dispatch(
+            addNotification({
+              type: 'error',
+              message: t(
+                'A category must be transferred to another of the same type (expense or income)',
+              ),
+            }),
+          );
+          break;
+        default:
+          dispatch(addGenericErrorNotification());
+      }
+
+      throw new Error(error);
+    } else {
+      await dispatch(getCategories());
```

Comment:
Don't we need to `.unwrap` all the `await`ed `dispatch` in this file so that errors will bubble up? Otherwise we'll no longer even see "unhandled promise rejection" or something in the logs

Reply:
If this is the case, we'll need to fix the instances from the previous PR as well.

Reply:
Ideally, we shouldn't be using `unwrap` too much. The most ideal approach would be to dispatch, then handle any failures on the reducers and let the component react to the state change.

I just put the `unwrap` calls here since some components depend on the result of some actions.

One thing we can do in the meantime while we haven't setup a more thorough error handling in the reducers is to catch the error in the `createAsyncThunk` and display an error notification - that way we don't pollute the `dispatch`s with `unwrap` calls. WDYT?